server/api.go

```diff
@@ -0,0 +1,316 @@
+	s.HandleFunc("/wopi/files/{fileID:[a-z0-9]+}/contents", p.saveWopiFileContents).Methods(http.MethodPost)
+
+	// 404 handler
+	r.Handle("{anything:.*}", http.NotFoundHandler())
+	return r
+}
+
+func (p *Plugin) getBaseAPIURL() string {
+	return *p.API.GetConfig().ServiceSettings.SiteURL + "/plugins/" + manifest.Id + "/api/v1"
+}
+
+// withRecovery allows recovery from panics
+func (p *Plugin) withRecovery(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if x := recover(); x != nil {
+				p.API.LogError("Recovered from a panic",
+					"url", r.URL.String(),
+					"error", x,
+					"stack", string(debug.Stack()))
+			}
+		}()
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+// handleStaticFiles handles the static files under the assets directory.
+func (p *Plugin) handleStaticFiles(r *mux.Router) {
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		p.API.LogWarn("Failed to get bundle path.", "Error", err.Error())
+		return
+	}
+
+	// This will serve static files from the 'assets' directory under '/static/<filename>'
+	r.PathPrefix("/static/").Handler(http.StripPrefix("/static/", http.FileServer(http.Dir(filepath.Join(bundlePath, "assets")))))
+}
+
+// handleAuthRequired verifies if provided request is performed by a logged-in Mattermost user.
+func handleAuthRequired(handleFunc func(w http.ResponseWriter, r *http.Request)) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		userID := r.Header.Get(HeaderMattermostUserID)
+		if userID == "" {
+			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			return
+		}
+
+		handleFunc(w, r)
+	}
+}
+
+// parseFileIDs sends the file info to the client (name, extension and id) for each file
+// body contains an array with file ids in JSON format
+func (p *Plugin) parseFileIDs(w http.ResponseWriter, r *http.Request) {
+	//extract fileIDs array from body
+	body, bodyReadError := ioutil.ReadAll(r.Body)
+	if bodyReadError != nil {
+		p.API.LogError("Error when reading body: ", bodyReadError.Error())
+		return
+	}
+	var fileIDs []string
+	_ = json.Unmarshal(body, &fileIDs)
+
+	//create an array with more detailed file info for each file
+	files := make([]ClientFileInfo, 0, len(fileIDs))
+	for _, fileID := range fileIDs {
+		fileInfo, fileInfoError := p.API.GetFileInfo(fileID)
+		if fileInfoError != nil {
+			p.API.LogError("Error when retrieving file info: ", fileInfoError.Error())
+			continue
+		}
+		if value, ok := WopiFiles[strings.ToLower(fileInfo.Extension)]; ok {
+			file := ClientFileInfo{
+				fileInfo.Id,
+				fileInfo.Name,
+				fileInfo.Extension,
+				value.Action,
+			}
+			files = append(files, file)
+		}
+	}
+
+	responseJSON, _ := json.Marshal(files)
+
+	w.Header().Set("Content-Type", "application/json")
+	if _, err := w.Write(responseJSON); err != nil {
+		p.API.LogError("failed to write status", "err", err.Error())
```

Comment:
I don't think "failed to write status" is the right error message. It should be like "failed to write response" or something else.